### ISSUE TITLE
User worker keeps connection open even after request was executed 

### ISSUE THREAD

**Issue by andreespirela** [supabase/edge-runtime#43]
# Bug report

On this line https://github.com/supabase/edge-runtime/blob/main/examples/main/index.ts#L23 . The timeout for the User Worker is defined (also known as the worker that processes the request).

Currently, when a request is sent and the user worker evaluates `serve`, and it returns the response back to the user, it seems like the connection is kept open and when the worker dies when killing the thread for timeout, the following message is logged in the console:

```
Error in user worker connection: error shutting down connection: Socket is not connected (os error 57)
```

You can verify this behavior by starting the main script and then calling `http://localhost:9000/main-empty` and then waiting for the timeout. Happening in PR #32 .

# Questions
- Do we want to keep the connection open or we want to kill the worker as soon as the response is proxied? The way GCP handles this for example is the connection is open until timeout is reached OR a response is given to the client (https://cloud.google.com/functions/docs/concepts/nodejs-runtime)

cc @laktek

**Comment by inian** [comment]
Is there a way to check if there are any tasks in the event loop? Ideally, we kill the worker if a response is returned and there are no pending tasks in the event loop. If there are pending tasks, we terminate after the timeout specified.

**Comment by andreespirela** [comment]
@inian I think all tasks should be awaited before a response is given. Bottom linea would be if an HTTP response is sent, execution should end. 

Though, how would we handle this with long serving serverless functions or SSE? Any proposals ? @inian @laktek

**Comment by laktek** [comment]
This should be fixed with the changes in #63

**Comment by laktek** [comment]
We want to reuse workers for multiple requests. So the user worker will run till the specified timeout. However, after each request socket between main and user worker is dropped after a request and need to be established again (it uses Unix sockets so less overhead compared to a TCP socket connection).
